src/main/java/parsing/Parser.java

```diff
@@ -14,21 +16,27 @@
 import java.io.FileWriter;
 import java.io.IOException;
 import java.io.InputStreamReader;
+import java.io.LineNumberReader;
 import java.io.OutputStreamWriter;
 
 /**
  * Created by 101010.
  */
-public final class Parser {
+public class Parser {
```

Comment:
Making this class final makes it very hard to test with mockito/powermock. Therefore I removed the final tag. However, the class should effectively still be final.